components/infobox/wikis/stormgate/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,249 @@
+	end
+
+	return widgets
+end
+
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:addCustomCells(widgets)
+	return {
+		Cell{
+			name = 'Approx. Winnings ' .. CURRENT_YEAR,
+			content = {_player.earningsPerYear[CURRENT_YEAR]}
+		},
+		Cell{
+			name = Abbreviation.make('Years Active', 'Years active as a player'),
+			content = {YearsActive.display({player = _player.pagename})
+		}
+		},
+		Cell{
+			name = Abbreviation.make('Years Active (caster)', 'Years active as a caster'),
+			content = {_player:_getActiveCasterYears()}
+		},
+	}
+end
+
+---@return string|nil
+function CustomPlayer:_getActiveCasterYears()
+	if Namespace.isMain() then
+		local queryData = mw.ext.LiquipediaDB.lpdb('broadcasters', {
+			query = 'year::date',
+			conditions = '[[page::' .. self.pagename .. ']] OR [[page::' .. self.pagename:gsub(' ', '_') .. ']]',
+			limit = 5000,
+		})
+
+		local years = Array.map(queryData, function(item) return tonumber(item.year_date) end)
+
+		return Table.isNotEmpty(years) and YearsActive.displayYears(years) or nil
+	end
+end
+
+---@return WidgetInjector
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+---@return Html?
+function CustomPlayer:createBottomContent()
+	if Namespace.isMain() then
+		return MatchTicker.participant({player = self.pagename})
+	end
+end
+
+---@param categories string[]
+---@return string[]
+function CustomPlayer:getWikiCategories(categories)
+	local args = self.args
+	for _, faction in pairs(self:readFactions(args.race)) do
+		table.insert(categories, Faction.toName(faction) .. ' Players')
+	end
+
+	return categories
+end
+
+---@param args table
+---@return string
+function CustomPlayer:nameDisplay(args)
+	local factions = self:readFactions(args.race)
+
+	local raceIcons = table.concat(Array.map(factions, function(faction)
+		return Faction.Icon{faction = faction, size = 'large'}
+	end))
+
+	local name = args.id or self.pagename
+
+	return raceIcons .. '&nbsp;' .. name
+end
+
+---@param race string?
+---@param asCategory boolean?
+---@return string
+function CustomPlayer:getRaceData(race, asCategory)
+	local factions = self:readFactions(race)
+
+	return table.concat(Array.map(factions, function(faction)
+		faction = Faction.toName(faction)
+		if asCategory then
+			return '[[:Category:' .. faction .. ' Players|' .. faction .. ']]'
+		end
+		return '[[' .. faction .. ']]'
+	end) or {}, ',&nbsp;')
+end
+
+---@param input string?
+---@return string[]
+function CustomPlayer:readFactions(input)
+	return Faction.readMultiFaction(input, {alias = false, sep = ','})
+end
+
+---@param lpdbData table<string, string|number|table|nil>
+---@param args table
+---@param personType string
+---@return table<string, string|number|table|nil>
+function CustomPlayer:adjustLPDB(lpdbData, args, personType)
+	local extradata = lpdbData.extradata or {}
+
+	local factions = self:readFactions(args.race)
+
+	extradata.faction = factions[1]
+	extradata.faction2 = factions[2]
```

Comment:
Is there players with multiple factions as default? Otherwise add it when it becoems relevant

Reply:
there will be for sure eventually